static/js/controllers/contacts-content.js

```diff
@@ -150,7 +153,10 @@ var _ = require('underscore'),
         return split[0].concat(split[1]);
       };
 
-      var isContactPrimaryContact = function(contactDoc) {
+      var isPersonAndPrimaryContact = function(contactDoc) {
+        if (contactDoc.type !== 'person') {
```

Comment:
Interesting. This is single-thread though. Is the point to limit the number of times you break off execution because of a promise call?